src/quest_manager/views.py

```diff
@@ -184,17 +184,25 @@ class QuestCopy(QuestCreate):
     def get_form_kwargs(self):
         kwargs = super().get_form_kwargs()
 
+        quest = get_object_or_404(Quest, pk=self.kwargs['quest_id'])
+
+        kwargs['initial'] = model_to_dict(quest)
+        kwargs['initial']['tags'] = quest.tags.all()
+
         # by default, set the quest this was copied from as the new_quest_prerequisite
         # If this is changed in the form it will be overwritten in form_valid() from QuestFormViewMixin
-        copied_quest = get_object_or_404(Quest, pk=self.kwargs['quest_id'])
-        kwargs['initial']['new_quest_prerequisite'] = copied_quest
+        kwargs['initial']['new_quest_prerequisite'] = quest
+
+        # rename copied quest
+        # strips front if len(name + ' - copy') is too long
+        char_limit = self.model._meta.get_field('name').max_length
```

Comment:
Not needed. If it is, let me know why (there are no code comments explaining....)

Reply:
removed

Reply:
removed